providers/base/bin/pipewire_test.py

```diff
@@ -0,0 +1,608 @@
+    """
+    parser = argparse.ArgumentParser(description=description)
+    parser.add_argument(
+            "-t", "--time",
+            dest='test_duration',
+            action='store',
+            default=30,
+            type=int,
+            help="""Maximum test duration, default %(default)s seconds.
+                    It may exit sooner if it determines it has enough data.""")
+    parser.add_argument(
+            "-a", "--audio",
+            action='store',
+            default="/dev/null",
+            type=str,
+            help="File to save recorded audio in .wav format")
+    parser.add_argument(
+            "-q", "--quiet",
+            action='store_true',
+            default=False,
+            help="Be quiet, no output unless there's an error.")
+    parser.add_argument(
+            "-d", "--debug",
+            action='store_true',
+            default=False,
+            help="Debugging output")
+    parser.add_argument(
+            "-f", "--frequency",
+            action='store',
+            default=DEFAULT_TEST_FREQUENCY,
+            type=int,
+            help="Frequency for test signal, default %(default)s Hz")
+    parser.add_argument(
+            "-u", "--spectrum",
+            action='store',
+            type=str,
+            help="""File to save spectrum information for plotting
+                    (one frequency/magnitude pair per line)""")
+    return parser.parse_args(args)
+
+
+def run_test(args):
+
+    # Setup logging
+    level = logging.INFO
+    if args.debug:
+        level = logging.DEBUG
+    if args.quiet:
+        level = logging.ERROR
+    logging.basicConfig(level=level)
+    try:
+        # Launches recording pipeline. I need to hook up into the gst
+        # messages.
+        recorder = Recorder(output_file=args.audio, logger=logging)
+        # Just launches the playing pipeline
+        player = Player(frequency=args.frequency, logger=logging)
+    except GObject.GError as excp:
+        logging.critical("Unable to initialize GStreamer pipelines: %s", excp)
+        sys.exit(127)
+
+    # This just receives a process feedback and tells me how much to change to
+    # achieve the setpoint
+    pidctrl = PIDController(kp=0.7, ki=.01, kd=0.01,
+                            setpoint=REC_LEVEL_RANGE[0])
+    pidctrl.set_change_limit(5)
+    # This  gathers spectrum data.
+    analyzer = SpectrumAnalyzer(points=BINS,
+                                sampling_frequency=SAMPLING_FREQUENCY)
+
+    recorder.volumecontroller = VolumeController(type='input',
+                                                 logger=logging)
+    recorder.volumecontroller.set_volume(0)
+    recorder.volumecontroller.mute(False)
+
+    player.volumecontroller = VolumeController(type='output',
+                                               logger=logging)
+    player.volumecontroller.set_volume(PLAY_VOLUME)
+    player.volumecontroller.mute(False)
```

Comment:
Why need the second mute?

Reply:
One is for `recorder` and other one is for `player`